osquery/tables/system/windows/programs.cpp

```diff
@@ -18,8 +18,104 @@
 namespace osquery {
 namespace tables {
 
+// Function to reverse a string
+std::string reverseString(const std::string& input) {
+  std::string reversed = input;
+  std::reverse(reversed.begin(), reversed.end());
+  return reversed;
+}
+
+// Function to convert a registry-encoded MSI UpgradeCode into a standard GUID
+std::string decodeMsiRegistryGuid(const std::string& encoded) {
+  // Ensure the encoded string is exactly 32 characters long
+  if (encoded.length() != 32) {
+    return "Invalid length";
+  }
+  // Microsoft uses a custom encoding for GUIDs in the registry
```

Comment:
Some of this seems to align with the native guid implementation in c++
https://learn.microsoft.com/en-us/windows/win32/api/guiddef/ns-guiddef-guid
```
typedef struct _GUID {
    unsigned long  Data1;
    unsigned short Data2;
    unsigned short Data3;
    unsigned char  Data4[8];
} GUID;
```
```
Data1: Represents the first 8 hexadecimal digits (32 bits).
Data2: Represents the next 4 hexadecimal digits (16 bits).
Data3: Represents the subsequent 4 hexadecimal digits (16 bits).
Data4: Represents the final 12 hexadecimal digits (64 bits), stored as an array of 8 bytes.
```